base/toml_parser.jl

```diff
@@ -899,10 +899,71 @@ function parse_float(l::Parser, contains_underscore)::Err{Float64}
     return v
 end
 
-function parse_int(l::Parser, contains_underscore, base=nothing)::Err{Int64}
+function parse_int(l::Parser, contains_underscore, base=nothing)::Err{Union{Base.BitSigned, BigInt}}
```

Comment:
Instead of having four almost identical functions how about
```julia
for (name, T1, T2, n1, n2) in (("int", Int64,  Int128,  17,  33),
                               ("hex", UInt64, UInt128, 18,  34),
                               ("oct", UInt64, UInt128, 24,  45),
                               ("bin", UInt64, UInt128, 66, 130),
                               )
    @eval function $(Symbol("parse_", name))(l::Parser, contains_underscore, base=nothing)::Err{Union{$(T1), $(T2), BigInt}}
        s = take_string_or_substring(l, contains_underscore)
        len = length(s)
        v = try
            if len ≤ $(n1)
                Base.parse($(T1), s; base)
            elseif $(n1) < len ≤ $(n2)
                Base.parse($(T2), s; base)
            else
                Base.parse(BigInt, s; base)
            end
        catch e
            e isa Base.OverflowError && return(ParserError(ErrOverflowError))
            error("internal parser error: did not correctly discredit $(repr(s)) as an int")
        end
        return v
    end
end
```
?